contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -4877,16 +4854,41 @@ transform_pivot_clause(ParseState *pstate, SelectStmt *stmt)
 	c_sql->stmt_location = 0;
 	c_sql->stmt_len = 0;
 
-	oldContext = MemoryContextSwitchTo(tsql_outmost_context);
-	/* save rewrited sqls to global variable for later retrive */
-	pivot_fields = (tsql_pivot_fields *) palloc(sizeof(tsql_pivot_fields));
-	pivot_fields->s_sql = copyObject(s_sql);
-	pivot_fields->c_sql = copyObject(c_sql);
-	pivot_fields->sourcetext = pstrdup(pstate->p_sourcetext);
-	pivot_fields->funcName = pstrdup(funcName->sval);
-	tsql_outmost_estat->pivot_parsetree_list = lappend(tsql_outmost_estat->pivot_parsetree_list, pivot_fields);
-	tsql_outmost_estat->pivot_number++;	
-	MemoryContextSwitchTo(oldContext);
+	pivot_context_list = list_make3(list_make1(makeString("bbf_pivot_func")),
+									list_make2((Node *) copyObject(s_sql),
+												(Node *) copyObject(c_sql)
+												),
+									list_make2(makeString(pstrdup(pstate->p_sourcetext)),
+												makeString(pstrdup(funcName->sval))
+												)
+									);
```

Comment:
We decided to not to introduce a new BbfPivotData node because it will cause unexpected test failures. Due to time limit, we choose to use list to store pivot data. The first string list is the identifier. We could possibly use ExtensibleNode in the next release which will limit the code change only to the extension.